docs/setup/sso.md

```diff
@@ -0,0 +1,298 @@
+    - id: "01HFS67GJ145HCM9ZASYS9DC3J"
+      issuer: "https://gitlab.com"
+      human_name: "GitLab"
+      brand_name: "gitlab"
+      token_endpoint_auth_method: "client_secret_post"
+      client_id: "<client-id>" # TO BE FILLED
+      client_secret: "<client-secret>" # TO BE FILLED
+      scope: "openid profile email"
+      claims_imports:
+        displayname:
+          action: suggest
+          template: "{{ user.name }}"
+        localpart:
+          action: ignore
+        email:
+          action: suggest
+          template: "{{ user.email }}"
+```
+
+
+### Google
+
+1. Set up a project in the Google API Console (see [documentation](https://developers.google.com/identity/protocols/oauth2/openid-connect#appsetup))
+2. Add an "OAuth Client ID" for a Web Application under ["Credentials"](https://console.developers.google.com/apis/credentials)
+3. Add the following "Authorized redirect URI": `https://<auth-service-domain>/upstream/callback/<id>`
+
+Authentication service configuration:
+
+```yaml
+upstream_oauth2:
+  providers:
+    - id: 01HFS6S2SVAR7Y7QYMZJ53ZAGZ
+      human_name: Google
+      brand_name: "google"
+      issuer: "https://accounts.google.com"
+      client_id: "<client-id>" # TO BE FILLED
+      client_secret: "<client-secret>" # TO BE FILLED
+      scope: "openid profile email"
+      claims_imports:
+        localpart:
+          action: ignore
+        displayname:
+          action: suggest
+          template: "{{ user.name }}"
+        email:
+          action: suggest
+          template: "{{ user.email }}"
+```
+
+
+### Keycloak
+
+
+Follow the [Getting Started Guide](https://www.keycloak.org/guides) to install Keycloak and set up a realm.
+
+1. Click `Clients` in the sidebar and click `Create`
+2. Fill in the fields as below:
+
+   | Field | Value |
+   |-----------|-----------|
+   | Client ID | `matrix-authentication-service` |
+   | Client Protocol | `openid-connect` |
+
+3. Click `Save`
+4. Fill in the fields as below:
+
+   | Field | Value |
+   |-----------|-----------|
+   | Client ID | `matrix-authentication-service` |
+   | Enabled | `On` |
+   | Client Protocol | `openid-connect` |
+   | Access Type | `confidential` |
+   | Valid Redirect URIs | `https://<auth-service-domain>/upstream/callback/<id>` |
+
+5. Click `Save`
+6. On the Credentials tab, update the fields:
+
+   | Field | Value |
+   |-------|-------|
+   | Client Authenticator | `Client ID and Secret` |
+
+7. Click `Regenerate Secret`
+8. Copy Secret
+
+```yaml
+upstream_oauth2:
+  providers:
+    - id: "01H8PKNWKKRPCBW4YGH1RWV279"
+      issuer: "https://<keycloak>/realms/<realm>" # TO BE FILLED
+      token_endpoint_auth_method: client_secret_basic
+      client_id: "matrix-authentication-service"
+      client_secret: "<client-secret>" # TO BE FILLED
+      scope: "openid profile email"
+      claims_imports:
+        localpart:
+          action: require
+          template: "{{ user.preferred_username }}"
+        displayname:
```

Comment:
In some sections it's called `displayname`, in others it's called `display_name`. Should we be consistent with the naming?

Reply:
Good catch, should be `displayname` everywhere